### DISCUSSION TITLE
Distinguish actors in the conversation

### DISCUSSION THREAD

**Discussion by atolmachev** [ggml-org/whisper.cpp#450] [General]
Is there a feature to distinguish actors (person1, person2, ...) so that it will be easier to use the output as the conversation/podcast/coaching session/... transcription?

**Comment by elijahfhopp** [comment]
Apparently this is called diarizaition. It appears to be available for stereo recordings in the command line:
```
-di,      --diarize         [false  ] stereo audio diarization
```
I haven't used it myself, so I can't vouch for its quality. 

If you have multi-track audio then you can obviously just use it on each track and then perhaps have a script that takes the timestamps to make them into one, cohesive transcription. I am currently working on a project that will use a system like this, so if you end up writing anything, ping me.

**Reply by strangelearning** [reply]
Anecdotally, a user and I have tried the same thing and diarization isn't quite working. 

https://github.com/ggerganov/whisper.cpp/issues/489

We have both converted mono audio files into stereo files using ffmpeg `ffmpeg -i input.wav -ac 2 output.wav`
 
And we both get (speaker ?) for our attempts. I'll be tracking the issue linked above.

The commands I am using are `./main -m models/ggml-tiny.bin  -f ../../openai/yt-semantic-search/firstep/outputstereo.wav  -di` and `./main -m models/ggml-tiny.bin  -f ../../openai/yt-semantic-search/firstep/outputstereo.wav  -di -ml 1` respectively.

<img width="1045" alt="Screenshot 2023-02-14 at 11 02 17 AM" src="https://user-images.githubusercontent.com/80677888/218791483-4e4e35d6-0f92-417a-9428-29e1a155733a.png">

**Reply by beschuldigung** [reply]
is there any way to make it so theres something like Speaker 1, speaker 2, etc?
